graphprot/NeuralNet.py

```diff
@@ -97,25 +97,65 @@ def __init__(self, database, Net,
             self.loss = nn.CrossEntropyLoss(weight = self.class_weights, reduction='mean')       
 
 
-    def train(self, nepoch=1, validate=False):
+    def train(self, nepoch=1, validate=False, plot=False):
+        
+        _train_acc = []
+        _train_loss = []
+        _valid_acc = []
+        _valid_loss = []
 
-        self.model.train()
         for epoch in range(1, nepoch+1):
+            self.model.train()
```

Comment:
why do you put train() in the loop  ? Do you change it somewhere in the iteration ?